connect/src/com/telenor/connect/ConnectSdk.java

```diff
@@ -657,6 +730,23 @@ public void onAvailable(Network network) {
         }
     }
 
+    private static void initializeAdvertisingId() {
+        GoogleApiAvailability googleAPI = GoogleApiAvailability.getInstance();
+        if (googleAPI.isGooglePlayServicesAvailable(getContext()) == ConnectionResult.SUCCESS) {
+            new Thread(new Runnable() {
+                public void run() {
+                    AdvertisingIdClient.Info adInfo = null;
+                    try {
+                        adInfo = AdvertisingIdClient.getAdvertisingIdInfo(getContext());
+                        advertisingId = adInfo.getId();
+                    } catch (Exception e) {
+                        Log.e(ConnectUtils.LOG_TAG, "Failed to read advertisment id", e);
```

Comment:
If it is possible that this will fail due to blocked permissions/settings (I am not sure here), then maybe we should just log with a warning instead of error?